src/Eloquent/Builder.php

```diff
@@ -71,7 +73,10 @@ public function update(array $values, array $options = [])
             return 1;
         }
 
-        return $this->toBase()->update($this->addUpdatedAtColumn($values), $options);
+        $builder = $this->toBase();
+        assert($builder instanceof MongoDBQueryBuilder);
+
+        return $builder->update($this->addUpdatedAtColumn($values), $options);
```

Comment:
So this is an interesting case. Both the IDE and PHPStan think that `Illuminate\Database\Query\Builder` is used, while actually it's `MongoDB\Laravel\Query\Builder`.
Adding the assertion is quickest/eastiest way to make sure the correct one is used.

Reply:
You can overload the return type of `toBase` with an annotation on this class.
```php
/** @method \MongoDB\Laravel\Query\Builder toBase() */
class Builder extends EloquentBuilder
```